### PR TITLE
migrate DropdownSetting component from class based to function based component

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

Migrate `DropdownSetting` component from class based to function based component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24750

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 1]
```diff
@@ -20,57 +19,51 @@ type Props = {
...
+        value,
+        onChange,
+        disabled = false,
+        setByEnv,
+        helpText,
+    } = props;
 
-    handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
-        this.props.onChange(this.props.id, e.target.value);
+    const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
```
Comment (jespino) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:34]: Can we put this in a `useCallback`?

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 2]
```diff
@@ -20,57 +19,51 @@ type Props = {
...
-            options.push(
-                <option
-                    value={value}
-                    key={value}
-                >
-                    {text}
-                </option>,
-            );
-        }
+    const options = useMemo(() =>
```
Comment (jespino) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:38]: Nice improvement! :)

**Review by jespino — CHANGES_REQUESTED** [review]
I tiny improvement suggestion, but it looks great. Thanks @tanmaythole! :fireworks:

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx (hunk 1 above)
Comment (tanmaythole) [reply webapp/channels/src/components/admin_console/dropdown_setting.tsx]: 👍🏻

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 3]
```diff
@@ -19,21 +19,19 @@ type Props = {
...
-        label,
-        value,
-        onChange,
-        disabled = false,
-        setByEnv,
-        helpText,
-    } = props;
-
-    const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
+const DropdownSetting: FC<Props> = ({
```
Comment (jespino) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:22]: Nice!

**Review by jespino — APPROVED** [review]
Now looks good to me! Thanks @tanmaythole! :notes:

File: webapp/channels/src/components/admin_console/dropdown_setting.tsx [hunk 4]
```diff
@@ -20,57 +19,49 @@ type Props = {
...
-        id: PropTypes.string.isRequired,
-        values: PropTypes.array.isRequired,
-        label: PropTypes.node.isRequired,
-        value: PropTypes.string.isRequired,
-        onChange: PropTypes.func.isRequired,
-        disabled: PropTypes.bool,
-        setByEnv: PropTypes.bool.isRequired,
-        helpText: PropTypes.node,
-    };
+const DropdownSetting: FC<Props> = ({
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/dropdown_setting.tsx:22]: The explicit type definition should not be needed. You can do something like `const Component = ({...}:Props) => {...}`.

**Review by larkox — APPROVED** [review]
LGTM! Just a minor nitpick. Approving so we can move into QA, but it would be great if you could change it :smile:

**Comment by larkox** [comment]
@tanmaythole There seem to be some problems with the tests. Just some snapshots not passing. It should be solved by running `npm run test -- -u -- path/to/test/file.test.ts` in the channels folder.

Here the details: https://github.com/mattermost/mattermost/actions/runs/6439967230/job/17488367246?pr=24796

**Review by lindy65 — APPROVED** [review]
Thanks @tanmaythole - looks good to me 👍 

cc @larkox

**Comment by larkox** [comment]
Thank you for the contribution!

**Pull request merged**